backends/gstreamer/src/player.rs

```diff
@@ -0,0 +1,397 @@
+use glib::*;
+use gst;
+use gst_app;
+use gst_player;
+use gst_player::PlayerStreamInfoExt;
+use ipc_channel::ipc;
+use servo_media_player::frame::{Frame, FrameRenderer};
+use servo_media_player::metadata::Metadata;
+use servo_media_player::{Player, PlaybackState, PlayerEvent};
+use std::string;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::time;
+use std::u64;
+
+struct GStreamerFrame {}
+
+impl GStreamerFrame {
+    fn new(sample: &gst::Sample) -> Frame {
+        let caps = sample.get_caps().unwrap();
+        let s = caps.get_structure(0).unwrap();
+        let width = s.get("width").unwrap();
+        let height = s.get("height").unwrap();
+        let buffer = sample.get_buffer().unwrap();
+
+        let map = buffer.map_readable().unwrap();
+        let data = Vec::from(map.as_slice());
+
+        Frame::new(width, height, Arc::new(data))
+    }
+}
+
+struct GStreamerMetadata {}
```

Comment:
ditto, this should just be a function like `create_metadata()`

Reply:
Alternatively, this can be a `From` impl, `impl From<&PlayerMediaInfo> for Metadata`